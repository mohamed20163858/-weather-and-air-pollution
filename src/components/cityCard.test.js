import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import CityCard from './cityCard';

function TestCities() {
  const id = 1;
  const country = 'Egypt';
  const city = 'Cairo';
  const iso2 = 'EG';
  const iso3 = 'EGY';
  return (
    <Provider store={store}>
      <Router>
        <CityCard
          countryName={country}
          cityName={city}
          cityISO2={iso2}
          cityISO3={iso3}
          id={id}
          key={`${country}-${city}`}
        />
      </Router>
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestCities />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('simulate showing card details ', () => {
  // ARRANGE
  render(<TestCities />);

  // ACT
  userEvent.click(screen.getByTestId('check-id'));
  // Arrange
  const result = screen.getByText('Country: Egypt');
  // ASSERT
  expect(result).toBeInTheDocument();
});
