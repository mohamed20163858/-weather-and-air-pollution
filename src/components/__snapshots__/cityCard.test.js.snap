// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="city-card dark"
  data-testid="check-id"
  href="/EG/Cairo"
  onClick={[Function]}
>
  <div
    id={1}
  >
    <img
      alt="Egypt flag"
      src="not_available.jpg"
    />
    <p>
      Country: Egypt
    </p>
    <p>
      City: Cairo
    </p>
  </div>
</a>
`;
